exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -285,8 +648,18 @@ public AggOutcome doWork() {
       // In the future HashAggregate may also need to perform some actions conditionally
       // in the outer try block.
 
+      assert ! handlingSpills || currentIndex < Integer.MAX_VALUE;
+
       outside:
       while (true) {
+
+        // This would be called only once - after actual data arrives on incoming
+        if ( schema == null && incoming.getRecordCount() > 0 ) {
+          this.schema = incoming.getSchema();
+          // Calculate the number of partitions based on actual incoming data
+          delayedSetup();
+        }
+
```

Comment:
Comment on original code below: no need for the EXTRA_DEBUG_1 and EXTRA_DEBUG_2: just use trace level logging and avoid the extra complexity.